packages/array-range/index.js

```diff
@@ -0,0 +1,30 @@
+module.exports = range;
+
+/*
+  range(0, 5); // [0, 1, 2, 3, 4]
+  range(5); // [0, 1, 2, 3, 4]
+  range(-5); // [0, -1, -2, -3, -4]
+  range(0, 20, 5) // [0, 5, 10, 15]
+  range(0, -20, -5) // [0, -5, -10, -15]
+*/
+
+function range(start, stop, step) {
+  if (typeof stop === 'undefined') {
+    stop = start || 0;
+    start = 0;
+  }
+  if (typeof step === 'undefined') {
+    step = stop > start ? 1 : -1;
+  }
+  var toReturn = [];
+  if (start < stop) {
+    for (; start < stop; start += step) {
+      toReturn.push(start);
+    }
+  } else {
+    for (; start > stop; start += step) {
+      toReturn.push(start);
+    }
+  }
```

Comment:
If start were assigned to the lower value always, the following would no longer be true:

```js
  var start = 0;
  var negativeStop = -5;
  t.deepEqual(range(start, negativeStop), [0, -1, -2, -3, -4]);
```

If I understand your proposition correctly, this would be the result

```js
  var start = 0;
  var negativeStop = -5;
  t.deepEqual(range(start, negativeStop), [-4, -3, -2, -1, 0]);
```

If this module seeks parity with other popular range functions available on npm, the first option should be the standard.